common/script/fns/ultimateGear.js

```diff
@@ -1,33 +1,35 @@
 import content from '../content/index';
 import _ from 'lodash';
 
-module.exports = function(user) {
-  var base, owned;
-  owned = typeof window !== "undefined" && window !== null ? user.items.gear.owned : user.items.gear.owned.toObject();
-  if ((base = user.achievements).ultimateGearSets == null) {
-    base.ultimateGearSets = {
+module.exports = function ultimateGear (user) {
+  let owned = window ? user.items.gear.owned : user.items.gear.owned.toObject();
+
+  if (!user.achievements.ultimateGearSets) {
+    user.achievements.ultimateGearSets = {
       healer: false,
       wizard: false,
       rogue: false,
-      warrior: false
+      warrior: false,
     };
   }
-  content.classes.forEach(function(klass) {
+
+  content.classes.forEach((klass) => {
     if (user.achievements.ultimateGearSets[klass] !== true) {
-      return user.achievements.ultimateGearSets[klass] = _.reduce(['armor', 'shield', 'head', 'weapon'], function(soFarGood, type) {
-        var found;
-        found = _.find(content.gear.tree[type][klass], {
-          last: true
+      user.achievements.ultimateGearSets[klass] = _.reduce(['armor', 'shield', 'head', 'weapon'], (soFarGood, type) => {
+        let found = _.find(content.gear.tree[type][klass], {
+          last: true,
         });
         return soFarGood && (!found || owned[found.key] === true);
       }, true);
     }
   });
-  if (typeof user.markModified === "function") {
-    user.markModified('achievements.ultimateGearSets');
-  }
+
+  // TODO
+  if (user.markModified) user.markModified('achievements.ultimateGearSets');
```

Comment:
@crookedneighbor I'm probably changing my stance about moving markModified out of the shared code... it's a mess when it's not called directly by an ops but from a nested function (ops.buy -> ultimateGear -> markModified)